services/frontend-service/src/ui/ConfirmationDialog.tsx

```diff
@@ -14,45 +14,24 @@ You should have received a copy of the GNU General Public License
 along with kuberpult.  If not, see <http://www.gnu.org/licenses/>.
 
 Copyright 2021 freiheit.com*/
-import { BatchAction, Lock } from '../api/api';
+import { Lock } from '../api/api';
 import * as React from 'react';
-import { Button, Dialog, DialogTitle, IconButton, Typography, Snackbar, Alert, AlertTitle } from '@material-ui/core';
 import { useCallback, useContext } from 'react';
-import {
-    Close,
-    DeleteForeverRounded,
-    DeleteOutlineRounded,
-    Error,
-    LockOpenRounded,
-    LockRounded,
-    MoveToInboxRounded,
-} from '@material-ui/icons';
+import { Alert, AlertTitle, Button, Dialog, DialogTitle, IconButton, Snackbar, Typography } from '@material-ui/core';
+import { Close, LockRounded } from '@material-ui/icons';
 import { ActionsCartContext } from './App';
+import { ActionTypes, CartAction, getActionDetails, isDeployAction } from './ActionDetails';
 
-enum ActionTypes {
-    Deploy,
-    PrepareUndeploy,
-    Undeploy,
-    CreateEnvironmentLock,
-    DeleteEnvironmentLock,
-    CreateApplicationLock,
-    DeleteApplicationLock,
-    UNKNOWN,
-}
-
-const inCart = (actions: BatchAction[], action: BatchAction) =>
-    actions ? actions.find((act) => JSON.stringify(act.action) === JSON.stringify(action.action)) : false;
+const inCart = (actions: CartAction[], action: CartAction) =>
+    actions ? actions.find((act) => JSON.stringify(act) === JSON.stringify(action)) : false;
 
-const isDeployment = (t: ActionTypes) =>
-    t === ActionTypes.Deploy || t === ActionTypes.PrepareUndeploy || t === ActionTypes.Undeploy;
-
-const getCartConflicts = (cartActions: BatchAction[], newAction: BatchAction) => {
-    const conflicts = new Set<BatchAction>();
+const getCartConflicts = (cartActions: CartAction[], newAction: CartAction) => {
+    const conflicts = new Set<CartAction>();
     for (const action of cartActions) {
```

Comment:
Note to self: I think here we are reducing over `cartActions` to produce `conflicts`. (i.e. the whole section could be simplified a lot by writing it in a functional way)

Reply:
yeah that's a good idea, i'll leave that for the next PR